server/routes/api/messages.js

```diff
@@ -43,4 +44,36 @@ router.post("/", async (req, res, next) => {
   }
 });
 
+router.put("/read", async (req, res, next) => {
+  try {
+    if (!req.user) {
+      return res.sendStatus(401);
+    }
+    const userId = req.user.id;
+    const { message } = req.body;
+
+    const conversation = await Conversation.findConversation(
+      message.senderId, userId
+    );
+    // if user is part of conversation 
+    if (conversation) {
+      // update message
+      const updatedMessage = await Message.update({ read: true }, {
+        where: { 
+          conversationId: conversation.id,
+          senderId: message.senderId,
+          read: 0,
+          id: {
+            [Op.lte]: message.id
```

Comment:
I'm curious about the reason for this. I think it's safe to assume that if someone opens the conversation we can just consider all the messages as read at that point

Reply:
My thinking here was, if we were sending out notifications (ie. APN), we could have an APN action that says something like "Mark as Read" without opening the conversation. Good catch, I believe this could be considered pre-mature optimization. Will update.